src/Auth0.AuthenticationApi/Builders/AuthorizationUrlBuilder.cs

```diff
@@ -1,153 +1,165 @@
+            return WithValue("audience", audience);
         }
 
         /// <summary>
-        /// Adds a new nonce query string parameter.
+        /// Adds the `nonce` query string parameter specifying a cryptographically random nonce.
         /// </summary>
         /// <param name="nonce">The nonce.</param>
-        /// <returns>The <see cref="AuthorizationUrlBuilder"/>.</returns>
+        /// <returns>Current <see cref="AuthorizationUrlBuilder"/> to allow fluent configuration.</returns>
+        /// <remarks>See https://auth0.com/docs/api-auth/tutorials/nonce for more details</remarks>
         public AuthorizationUrlBuilder WithNonce(string nonce)
         {
-            AddQueryString("nonce", nonce);
-
-            return this;
+            return WithValue("nonce", nonce);
         }
 
         /// <summary>
-        /// Adds a new response_mode query string parameter.
+        /// Adds the `response_mode` query string parameter.
         /// </summary>
         /// <param name="responseMode">The response mode.</param>
         /// <returns>The <see cref="AuthorizationUrlBuilder"/>.</returns>
         public AuthorizationUrlBuilder WithResponseMode(AuthorizationResponseMode responseMode)
         {
-            AddQueryString("response_mode", AuthorizationResponseModeHelper.ConvertToString(responseMode));
-
-            return this;
+            return WithValue("response_mode", AuthorizationResponseModeHelper.ConvertToString(responseMode));
         }
 
         /// <summary>
-        /// Adds a new connection_scope query string parameter.
+        /// Adds the `connection_scope` query string parameter.
         /// </summary>
-        /// <param name="connectionScope">The connection scope to be passed to the corresponding connection. Multiple scopes must be separated by a space character.</param>
-        /// <returns>The <see cref="AuthorizationUrlBuilder"/>.</returns>
+        /// <param name="connectionScope">Connection scope to be passed to the corresponding connection. Multiple scopes must be separated by a space character.</param>
+        /// <returns>Current <see cref="AuthorizationUrlBuilder"/> to allow fluent configuration.</returns>
         public AuthorizationUrlBuilder WithConnectionScope(string connectionScope)
         {
-            AddQueryString("connection_scope", connectionScope);
+            return WithValue("connection_scope", connectionScope);
+        }
 
-            return this;
+        /// <summary>
+        /// Adds the `connection_scope` query string parameter.
+        /// </summary>
+        /// <param name="connectionScope">Connection scopes to be passed to the corresponding connection.</param>
+        /// <returns>Current <see cref="AuthorizationUrlBuilder"/> to allow fluent configuration.</returns>
+        public AuthorizationUrlBuilder WithConnectionScopes(params string[] connectionScope)
+        {
+            return WithConnectionScope(String.Join(" ", connectionScope));
         }
```

Comment:
`max_age` setter missing. Applies to this or not?

Reply:
It wasn't there before do you think there is value?

Reply:
Let's add it in a future update/PR - nobody has asked for it yet and it wouldn't be a breaking change to add it later.